pex/interpreter.py

```diff
@@ -183,10 +184,23 @@ def get(cls, binary=None):
         # vendored attrs distribution so that its `cache_hash=True` feature can work (see the
         # bottom of pex/third_party/__init__.py where the vendor importer is installed). We ignore
         # such adjoined `sys.path` entries to discover the true base interpreter `sys.path`.
-        pythonpath = frozenset(
-            os.environ.get("PYTHONPATH", "").split(os.pathsep) + list(third_party.exposed())
+        pythonpath = os.environ.get("PYTHONPATH")
+        internal_entries = frozenset(
+            (pythonpath.split(os.pathsep) if pythonpath else []) + list(third_party.exposed())
         )
-        sys_path = OrderedSet(item for item in sys.path if item and item not in pythonpath)
+
+        def is_internal_entry(entry):
+            # type: (str) -> bool
+            if entry in internal_entries:
+                return True
+            if not os.path.isabs(entry):
+                return False
+            for internal_entry in internal_entries:
+                if internal_entry == commonpath((internal_entry, entry)):
+                    return True
+            return False
+
+        sys_path = OrderedSet(entry for entry in sys.path if entry and not is_internal_entry(entry))
```

Comment:
Just confirming that both this `sys.path` and the `internal_entries` lists will be relatively small, so the nested loops are unlikely to be a performance problem?

Reply:
Correct. But even so, the cross product must be checked anyhow. There is no clever shortcut I can see.

Reply:
If it _was_ a problem, I have a suspicion could build a trie from the components of `internal_entries`, and traverse it using the components of `entry`, but much better if we can avoid that complexity.